src/shared/components/modal/createModalLauncher.tsx

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import { Modal, ModalProps } from '@patternfly/react-core';
+
+export type ModalComponentProps = Omit<ModalProps, 'children' | 'isOpen' | 'appendTo'> & {
+  ref?: React.LegacyRef<Modal>;
+};
+
+export type ModalLauncher = (onClose: () => void) => React.ReactElement;
+
+export type CreateModalLauncher = <P extends ModalComponentProps>(
+  C: React.ComponentType<P>,
+  args?: ModalComponentProps & {
```

Comment:
Probably better name to make it clear 
```suggestion
  modalConfig?: ModalComponentProps & {
```